ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java

```diff
@@ -0,0 +1,397 @@
+          }
+          console
+              .printError("Execution has failed. stack trace: " + ExceptionUtils.getStackTrace(e));
+          rc = 1;
+          done = true;
+        } else {
+          console.printInfo("Retrying...");
+        }
+      } finally {
+        if (done) {
+          if (rc != 0 && status != null) {
+            for (String diag : status.getDiagnostics()) {
+              console.printError(diag);
+              diagnostics.append(diag);
+            }
+          }
+          synchronized (shutdownList) {
+            shutdownList.remove(dagClient);
+          }
+          break;
+        }
+      }
+    }
+
+    perfLogger.PerfLogEnd(CLASS_NAME, PerfLogger.TEZ_RUN_DAG);
+    printSummary(success, vertexProgressMap);
+    return rc;
+  }
+
+  private void printSummary(boolean success, Map<String, Progress> progressMap) {
+    if (isProfilingEnabled() && success && progressMap != null) {
+
+      double duration = (System.currentTimeMillis() - this.executionStartTime) / 1000.0;
+      console.printInfo("Status: DAG finished successfully in " + String.format("%.2f seconds", duration));
+      console.printInfo("");
+
+      new QueryExecutionBreakdownSummary(perfLogger).print(console);
+      new DAGSummary(progressMap, hiveConf, dagClient, dag, perfLogger).print(console);
+
+      //llap IO summary
+      if (HiveConf.getBoolVar(hiveConf, HiveConf.ConfVars.LLAP_IO_ENABLED, false)) {
+        new LLAPioSummary(progressMap, dagClient).print(console);
+        new FSCountersSummary(progressMap, dagClient).print(console);
+      }
+      console.printInfo("");
+    }
+  }
+
+  private static boolean hasInterruptedException(Throwable e) {
+    // Hadoop IPC wraps InterruptedException. GRRR.
+    while (e != null) {
+      if (e instanceof InterruptedException || e instanceof InterruptedIOException) {
+        return true;
+      }
+      e = e.getCause();
+    }
+    return false;
+  }
+
+  /**
+   * killRunningJobs tries to terminate execution of all
+   * currently running tez queries. No guarantees, best effort only.
+   */
+  private static void killRunningJobs() {
+    synchronized (shutdownList) {
+      for (DAGClient c : shutdownList) {
+        try {
+          System.err.println("Trying to shutdown DAG");
+          c.tryKillDAG();
+        } catch (Exception e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  static long getCounterValueByGroupName(TezCounters vertexCounters, String groupNamePattern,
+                                         String counterName) {
+    TezCounter tezCounter = vertexCounters.getGroup(groupNamePattern).findCounter(counterName);
+    return (tezCounter == null) ? 0 : tezCounter.getValue();
+  }
+
+  private String updateStatus(DAGStatus status, Map<String, Progress> vertexProgressMap,
+      String lastReport) {
+    String report = getReport(vertexProgressMap);
+    if (!report.equals(lastReport) || System.currentTimeMillis() >= lastPrintTime + PRINT_INTERVAL) {
```

Comment:
This is showing some difference in behaviour in CLI (duration refreshes every 3 seconds). My guess is the duration was included in the report previously and report.equals(lastReport) fails every time printing the output. Not sure though. I will post screenshots to jira.